sdk/clientcore/annotation-processor/src/main/java/io/clientcore/annotation/processor/templating/JavaParserTemplateProcessor.java

```diff
@@ -103,12 +106,24 @@ public void process(TemplateInput templateInput, ProcessingEnvironment processin
         addImports(templateInput);
         addOrphanComments();
         setPackageDeclaration(packageName);
+        // Add class-level Javadoc instead of a line comment
+        compilationUnit.addOrphanComment(
+            new JavadocComment("This class is generated by annotation-processor. Initializes a new instance of the "
```

Comment:
A better option for this is setting the Javadoc on the `ClassOrInterfaceDeclaration` created in `createClass`.

```java
classBuilder = compilationUnit.addClass(serviceInterfaceImplShortName, Modifier.Keyword.PUBLIC)
    .setJavadocComment("This class is generated by annotation-processor. Initializes a new instance of the "...)
```

Also, might not be the best to just say `annotation-processor` as that's a concept in Java, maybe link to the GitHub source or Maven package?